R/count_teleconnections.R

```diff
@@ -55,6 +55,101 @@ count_watershed_teleconnections <- function(data_dir,
+
+      # Delete the rows that have crop types not present in CDL classes.
+      gcam_csv <- gcam_csv[-c(70, 100, 131, 45, 104, 182, 181),]
+
+      # Extract only the GCAM Class column
+      as.data.frame(gcam_csv[,c(3)]) %>%
+        mutate(GCAM_ID = group_indices_(gcam_csv, .dots=c("GCAM_commodity"))) ->
+        gcam_classes
+
+      # Shrink the Column so that there is only one of each Class
+      gcam_class <- distinct(gcam_classes)
+
+      # Obtain the attribute table for the raster.
+      levels(cropcover_USA)[[1]] %>%
+        filter(Class_Names != "") %>%
+        dplyr::select(ID, CDL_Class = Class_Names) %>%
+        mutate(CDL_Class = as.character(CDL_Class))->
+            cdl_classes
+
+      # This HUGE chunk assigns the GCAM IDs to matching CDL Class. Most likely a more efficient way of doing this.
+      # Since there were some differences in classes, I did the best I could to assign each class properly. All Double Crops are under MiscCrop.
+      cond1 <- cdl_classes$ID %in% c(1,12,13,225)
+        cdl_classes$GCAM_ID[cond1] <- 1
+      cond2 <- cdl_classes$ID %in% c(2,32,36)
+        cdl_classes$GCAM_ID[cond2] <- 2
+      cond3 <- cdl_classes$ID %in% c(59,60)
+        cdl_classes$GCAM_ID[cond3] <- 3
+      cond4 <- cdl_classes$ID %in% c(37,58,224)
+        cdl_classes$GCAM_ID[cond4] <- 4
+      cond5 <- cdl_classes$ID %in% c(10,11,14,42,44,47:57,61,66:72,74:77,204,206:210,212:223,227,229:254,226)
+        cdl_classes$GCAM_ID[cond5] <- 5
+      cond6 <- cdl_classes$ID %in% c(5,6,26,31,33:35,38,211)
+        cdl_classes$GCAM_ID[cond6] <- 6
+      cond7 <- cdl_classes$ID %in% c(4,21,25,27:29,39,205)
+        cdl_classes$GCAM_ID[cond7] <- 7
+      cond8 <- cdl_classes$ID %in% c(3)
+        cdl_classes$GCAM_ID[cond8] <- 8
+      cond9 <- cdl_classes$ID %in% c(43,46)
+        cdl_classes$GCAM_ID[cond9] <- 9
+      cond10 <- cdl_classes$ID %in% c(41,45)
+        cdl_classes$GCAM_ID[cond10] <- 10
+      cond11 <- cdl_classes$ID %in% c(22:24,30)
+        cdl_classes$GCAM_ID[cond11] <- 11
+      cond12 <- cdl_classes$ID %in% c(82,121:124)
+        cdl_classes$GCAM_ID[cond12] <- 12
+      cond13 <- cdl_classes$ID %in% c(87,190,195)
+        cdl_classes$GCAM_ID[cond13] <- 13
+      cond14 <- cdl_classes$ID %in% c(92)
+        cdl_classes$GCAM_ID[cond14] <- 14
+      cond15 <- cdl_classes$ID %in% c(141:143,63)
+        cdl_classes$GCAM_ID[cond15] <- 15
+      cond16 <- cdl_classes$ID %in% c(64,152:176)
+        cdl_classes$GCAM_ID[cond16] <- 16
+      cond17 <- cdl_classes$ID %in% c(112)
+        cdl_classes$GCAM_ID[cond17] <- 17
+      cond18 <- cdl_classes$ID %in% c(131,65)
+        cdl_classes$GCAM_ID[cond18] <- 18
+      cond19 <- cdl_classes$ID %in% c(81,88,83,111,0)
+        cdl_classes$GCAM_ID[cond19] <- 0
+
+
+      # Once all of the classes are assigned, this next part merges the GCAM Class name with the ID.
+      crop_reclass_table<- merge(cdl_classes, gcam_class, by = "GCAM_ID", all.x = TRUE)
```

Comment:
All hard-coded names (e.g., `GCAM_ID`) should be passed in as an argument to the function set with a default.